server/auto_updater/dataset_updater.py

```diff
@@ -0,0 +1,288 @@
+        )
+        dir_writer.writeheader()
+
+        embed_writer = csv.DictWriter(
+            embed_file,
+            delimiter="\t",
+            fieldnames=["journal", "document"] + [f"feat_{idx}" for idx in range(300)],
+        )
+        embed_writer.writeheader()
+
+        count_writer = csv.DictWriter(
+            word_file, delimiter="\t", fieldnames=["document", "lemma", "count"]
+        )
+        count_writer.writeheader()
+
+        # Cycle through each tar file on the server
+        for tar_file in tqdm.tqdm(tar_files):
+
+            # If not xml files skip
+            if all([suffix != ".xml" for suffix in Path(tar_file).suffixes]):
+                continue
+
+            # Grab the file from the tarfile
+            print(f"Requesting {pmc_open_access_url}{tar_file}....")
+            requested_file_stream = request.urlopen(f"{pmc_open_access_url}{tar_file}")
+            open_stream = tarfile.open(fileobj=requested_file_stream, mode="r|gz")
+
+            # Cycle through paper members
+            # Grab new papers and get the document vectors
+            while True:
+                try:
+
+                    pmc_paper = open_stream.next()
+
+                    if pmc_paper is None:
+                        break
+
+                    if pmc_paper.isdir():
+                        continue
+
+                    if pmc_paper.name in current_pmc_set:
+                        continue
+
+                    new_paper = open_stream.extractfile(pmc_paper)
+                    doc_vector, word_counter = generate_vector_counts(
+                        word_model, new_paper, "//abstract/sec/*|//body/sec/*"
+                    )
+
+                    if word_counter is None:
+                        continue
+
+                    dir_writer.writerow(
+                        {"tarfile": tar_file, "file_path": str(pmc_paper.name)}
+                    )
+
+                    embed_writer.writerow(
+                        {
+                            "document": Path(pmc_paper.name).stem,
+                            "journal": str(Path(pmc_paper.name).parent),
+                            **dict(
+                                zip([f"feat_{idx}" for idx in range(300)], doc_vector)
+                            ),
+                        }
+                    )
+
+                    for tok in word_counter:
+                        count_writer.writerow(
+                            {
+                                "document": Path(pmc_paper.name).stem,
+                                "lemma": tok,
+                                "count": word_counter[tok],
+                            }
+                        )
+
+                except tarfile.ReadError as e:
+                    print(
+                        f"File {pmc_open_access_url}{tar_file} achieved an error: {str(e)}"
+                    )
+                    print("Skipping...")
+                    break
```

Comment:
It seems to make more sense to use `continue` on this line (or delete this line completely), because the exception happens on the current `pmc_paper`, but we may still be able to process the other papers in this tarball file.